applications/tari_console_wallet/src/grpc/wallet_grpc_server.rs

```diff
@@ -175,6 +197,53 @@ impl wallet_server::Wallet for WalletGrpcServer {
 
         Ok(Response::new(GetTransactionInfoResponse { transactions }))
     }
+
+    async fn get_all_completed_transactions(
+        &self,
+        _request: Request<GetAllCompletedTransactionsRequest>,
+    ) -> Result<Response<GetTransactionInfoResponse>, Status>
+    {
+        debug!(
+            target: LOG_TARGET,
+            "Incoming GRPC request for GetAllCompletedTransactions"
+        );
+        let mut transaction_service = self.get_transaction_service();
+        let transactions;
+        match transaction_service.get_completed_transactions().await {
+            Ok(t) => {
+                transactions = t;
+            },
+            Err(_) => {
+                return Err(Status::not_found("No completed transactions found."));
+            },
+        };
+
+        let transactions = transactions
+            .into_iter()
+            .map(|tx| TransactionInfo {
+                tx_id: tx.1.tx_id,
+                source_pk: tx.1.source_public_key.to_vec(),
+                dest_pk: tx.1.destination_public_key.to_vec(),
+                status: TransactionStatus::from(tx.1.status) as i32,
+                amount: tx.1.amount.into(),
+                is_cancelled: tx.1.cancelled,
+                direction: TransactionDirection::from(tx.1.direction) as i32,
+                fee: tx.1.fee.into(),
+                timestamp: Some(naive_datetime_to_timestamp(tx.1.timestamp)),
+                excess_sig: tx
+                    .1
+                    .transaction
+                    .first_kernel_excess_sig()
+                    .expect("Complete transaction has no kernels")
+                    .get_signature()
+                    .to_vec(),
+                message: tx.1.message,
+                valid: tx.1.valid,
+            })
+            .collect();
+
+        Ok(Response::new(GetTransactionInfoResponse { transactions }))
+    }
 }
 
 fn convert_wallet_transaction_into_transaction_info(
```

Comment:
This is beyond the scope of this PR but this function duplicates the functionality of the following two implementations:
https://github.com/tari-project/tari/blob/development/base_layer/wallet/src/transaction_service/storage/models.rs#L312
https://github.com/tari-project/tari/blob/development/base_layer/wallet/src/transaction_service/storage/models.rs#L334

Seems unnecessary to maintain this in two places.

Reply:
Agreed - maybe for a future PR